lib/ffi-rzmq/message.rb

```diff
@@ -126,7 +126,7 @@ def copy_in_string string
     def copy_in_bytes bytes, len
       data_buffer = LibC.malloc len
       # writes the exact number of bytes, no null byte to terminate string
-      data_buffer.write_string bytes, len
+      data_buffer.put_bytes 0, bytes, 0, len
```

Comment:
There is also `write_bytes`, `data_buffer.write_bytes bytes, 0, len`.

If the intend is to always write has many bytes as `bytes` has (len==bytes.bytesize), then it can be simplified to:
`data_buffer.write_bytes bytes`

Reply:
Thx for the recommendation. I used put_bytes on purpose to guarantee the widest range of compatibility.

Reply:
Was there a FFI release where `put_bytes` was defined but `write_bytes` wasn't, or they behaved differently?

Reply:
I don't know nd don't have time to research this.